src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -2826,6 +2821,18 @@ public void psubscribe(final JedisPubSub jedisPubSub, final String... patterns)
     return params;
   }
 
+  @Override
+  public Object eval(final String script, final int keyCount, final String... params) {
+    checkIsInMultiOrPipeline();
+    client.eval(script, keyCount, params);
+    client.setTimeoutInfinite();
```

Comment:
don't you think it's better to move `client.setTimeoutInfinite();` into the try block?

Reply:
No. `setTimeoutInfinite()` has it's own try-catch. And if there is any exception, it is decided to be broken. There is no gain to `rollbackTimeout()` a broken connection.